torch/fx/experimental/efficient_conv_bn_eval.py

```diff
@@ -0,0 +1,143 @@
+
+    # shape of [C_out, 1, 1, 1] in Conv2d
+    weight_coeff = torch.rsqrt(bn.running_var + bn.eps).reshape(
+        [-1] + [1] * (conv.weight.ndim - 1)
+    )
+    # shape of [C_out, 1, 1, 1] in Conv2d
+    coefff_on_the_fly = bn_weight.view_as(weight_coeff) * weight_coeff
+
+    # shape of [C_out, C_in, k, k] in Conv2d
+    weight_on_the_fly = weight_on_the_fly * coefff_on_the_fly
+    # shape of [C_out] in Conv2d
+    bias_on_the_fly = bn_bias + coefff_on_the_fly.flatten() * (
+        bias_on_the_fly - bn.running_mean
+    )
+
+    bias_on_the_fly = bias_on_the_fly.to(dtype=x.dtype)
+    weight_on_the_fly = weight_on_the_fly.to(dtype=x.dtype)
+    return conv._conv_forward(x, weight_on_the_fly, bias_on_the_fly)
+
+
+def efficient_conv_bn_eval_control(
+    bn: nn.modules.batchnorm._BatchNorm, conv: nn.modules.conv._ConvNd, x: torch.Tensor
+):
+    """This function controls whether to use `efficient_conv_bn_eval_forward`.
+    If the following `bn` is in `eval` mode, then we turn on the special
+    `efficient_conv_bn_eval_forward`.
+    """
+    if not bn.training:
+        # bn in eval mode
+        output = efficient_conv_bn_eval_forward(bn, conv, x)
+        return output
+    else:
+        conv_out = conv._conv_forward(x, conv.weight, conv.bias)
+        return bn(conv_out)
+
+
+def efficient_conv_bn_eval_graph_transform(fx_model):
+    """Find consecutive conv+bn calls in the graph, inplace modify the graph
+    with the fused operation."""
+    modules = dict(fx_model.named_modules())
+
+    patterns = [(torch.nn.modules.conv._ConvNd, torch.nn.modules.batchnorm._BatchNorm)]
+
+    pairs = []
+    # Iterate through nodes in the graph to find ConvBN blocks
+    for node in fx_model.graph.nodes:
+        # If our current node isn't calling a Module then we can ignore it.
+        if node.op != "call_module":
+            continue
+        target_module = modules[node.target]
+        found_pair = False
+        for conv_class, bn_class in patterns:
+            if isinstance(target_module, bn_class):
+                target = node.args[0].target
+                if target not in modules:
+                    continue
+                source_module = modules[target]
+                if isinstance(source_module, conv_class):
+                    found_pair = True
+        # Not a conv-BN pattern or output of conv is used by other nodes
+        if not found_pair or len(node.args[0].users) > 1:
+            continue
+
+        # Find a pair of conv and bn computation nodes to optimize.
+        # To avoid modifying the graph during traversal, we record all pairs
+        # first, and then optimize them later.
+        conv_node = node.args[0]
+        bn_node = node
+        if modules[bn_node.target].track_running_stats:
+            pairs.append([conv_node, bn_node])
+
+    for conv_node, bn_node in pairs:
+        # set insertion point
+        fx_model.graph.inserting_before(conv_node)
```

Comment:
`with fx_model.graph.inserting_before(conv_node):`

Reply:
Fixed in [f113bc0](https://github.com/pytorch/pytorch/pull/108757/commits/f113bc06841e7671f7764bace1bf24cd59179b46).